apps/word-cloud/src/App.js

```diff
@@ -118,7 +134,29 @@ class App extends React.Component {
   };
 
   render() {
-    const { settings, context, words } = this.state;
+    const { words, hasNoBoards,  hasApiError } = this.state;
```

Comment:
instead of repeating the same div 3 times (lines 140, 151 and 161) I suggest making the decision of what to render outside of the render function and then rendering it here.

```
contentToRender = () => {
  const { words, hasNoBoards,  hasApiError } = this.state;
  if (hasNoBoards) {
    return (<AttentionBox 
            title="No boards connected" 
            text="Please connect a board to continue." 
            type={AttentionBox.types.DANGER}
          />)
  }
...
...
...
}

render() {
  const ContentToRender = this.contentToRender();
...
  return (<div className="monday-app">
          <ContentToRender />
        </div>)
}
```

Reply:
Aha nice - I knew my way was clunky, but wasn't sure how to resolve it. Thank you!